### DISCUSSION TITLE
[Feature Request]: T-Deck keyboard and screen lock

### DISCUSSION THREAD

**Discussion by joneyes** [meshtastic/firmware#3835] [Ideas]
### Platform

ESP32

### Description

Similar to Alt B toggling keyboard backlight...
Can a Alt L function be added to toggle lock on keyboard, trackball scrolling and screen?
In locked mode display padlock in corner of screen to indicate lock and press trackball to manually wake screen.

**Comment by emiliogon** [comment]
Please, please add this feature.  I'm tired of watching my T-Deck waste power by turning the screen on for unknown reasons, and I'm hesitant to carry it in a pocket because any key presses start sending a message (another UI issue).

I want an assertive, undeniable screen lockout that won't even wake for notifications.  I want the reset button to toggle it, because it has no practical use on a device with a power switch.

**Reply by mixpix** [reply]
I agree this feature would be very handy, if it's possible. It's too easy to send accidental messages. If it's not possible then I might have to look into designing a 3D printed cover or holster for the T-Deck Plus.

@emiliogon I believe 2.5.17 and 18 may have resolved the issue of the screen turning on randomly. My T-Deck Plus would turn the screen on randomly when it's just sitting on the desk in 2.5.15, after I updated to 18 I haven't seen the issue.
